src/Common/src/System/Net/Sockets/SocketErrorPal.Unix.cs

```diff
@@ -0,0 +1,145 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Net.Sockets
+{
+    internal static class SocketErrorPal
+    {
+#if DEBUG
+        static SocketErrorPal()
+        {
+            Debug.Assert(s_nativeErrorToSocketError.Count == NativeErrorToSocketErrorCount,
+                $"Expected s_nativeErrorToSocketError to have {NativeErrorToSocketErrorCount} count instead of {s_nativeErrorToSocketError.Count}.");
+
+            Debug.Assert(s_socketErrorToNativeError.Count == SocketErrorToNativeErrorCount,
+                $"Expected s_socketErrorToNativeError to have {SocketErrorToNativeErrorCount} count instead of {s_socketErrorToNativeError.Count}.");
+        }
+#endif
+
+        private const int NativeErrorToSocketErrorCount = 40;
+        private const int SocketErrorToNativeErrorCount = 40;
+
+        // No Interop.Errors are included for the following SocketErrors, as there's no good mapping:
+        // - SocketError.NoRecovery
+        // - SocketError.NotInitialized
+        // - SocketError.ProcessLimit
+        // - SocketError.SocketError
+        // - SocketError.SystemNotReady
+        // - SocketError.TypeNotFound
+        // - SocketError.VersionNotSupported
+
+        private static readonly Dictionary<Interop.Error, SocketError> s_nativeErrorToSocketError = new Dictionary<Interop.Error, SocketError>(NativeErrorToSocketErrorCount)
```

Comment:
Nit: personally, I prefer switch statements to the dictionary-based approach given here. Not blocking; either style is fine.